tests/stack/test_api.py

```diff
@@ -235,6 +243,15 @@ def test_deploy_job(self, stack_api):
         with pytest.raises(StackError):
             stack_api._deploy_job(alt_test_job_settings)
 
+        # TEST CASE 5
+        # If a databricks_id is deleted from the workspace while the status file contains that
```

Comment:
This test is more about inconsistency caused by "job not found", not about "deleted job". I would phrase the test to
"If a databricks_id is not found in workspace, then abort"
and then use variables such as 
`nonexisting_databricks_id`